src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1614,10 +1615,8 @@ private static async Task ApplyBuildArgumentsAsync(Container dcpContainerResourc
                 var valueString = buildSecret.Value switch
                 {
                     FileInfo filePath => filePath.FullName,
-                    string stringValue => stringValue,
                     IValueProvider valueProvider => await valueProvider.GetValueAsync(cancellationToken).ConfigureAwait(false),
-                    bool boolValue => boolValue ? "true" : "false",
-                    _ => buildSecret.Value.ToString()
+                    _ => throw new InvalidOperationException("Build secret can only be a parameter or a file.")
```

Comment:
Oh that's nice

Reply:
"opinionated" to me means going out of our way to make sure people don't shoot themselves in their feet :)

Reply:
Makes https://github.com/dotnet/aspire/issues/4429 more important.